tests/integration/test_installation.py

```diff
@@ -39,9 +40,34 @@
 def new_installation(ws, sql_backend, env_or_skip, inventory_schema, make_random):
     cleanup = []
 
-    def factory(config_transform: Callable[[WorkspaceConfig], WorkspaceConfig] | None = None):
+    def factory(  # pylint: disable=too-many-locals
+        config_transform: Callable[[WorkspaceConfig], WorkspaceConfig] | None = None,
+        single_user_install: bool = False,
+        fresh_install: bool = True,
+        existing_installation_prefix: str = '',
+        force_prompt_confirmation='no',
+    ):
         prefix = make_random(4)
+
+        # hack to mock the product info
+        # pylint: disable=import-outside-toplevel
+        from databricks.labs.ucx import install
+
+        install.PRODUCT_INFO = MagicMock()
```

Comment:
can you refactor `PRODUCT_INFO` to be injected via constructor? it's only used in the `install.py` (as it should), so it's fine. Initialize it in the `if __name__ == "__main__":` block.

<img width="858" alt="image" src="https://github.com/databrickslabs/ucx/assets/259697/753f6bcf-7847-4ad6-91be-5d66c4c86a20">

and then for the integration tests, just give it a random product name